packages/core/src/app.ts

```diff
@@ -1476,3 +1493,58 @@ function combinePackageJSON(...layers: object[]) {
   }
   return mergeWith({}, ...layers, custom);
 }
+
+function macroBabelPlugins() {
+  const isProduction = process.env.EMBER_ENV === 'production';
+  const isDebug = !isProduction;
+  const ignore = {
+    '@ember/debug': ['assert', 'deprecate', 'warn'],
+    '@ember/application/deprecations': ['deprecate'],
+  };
+
+  return [
+    [
+      require.resolve('babel-plugin-ember-modules-api-polyfill'),
+      { ignore },
+      '@embroider/core:babel-plugin-ember-modules-api-polyfill',
+    ],
+    [
+      require.resolve('babel-plugin-debug-macros'),
+      {
+        flags: [
+          {
+            source: '@glimmer/env',
+            flags: { DEBUG: isDebug, CI: !!process.env.CI },
+          },
+        ],
+
+        externalizeHelpers: {
+          global: 'Ember',
+        },
+
+        debugTools: {
+          isDebug,
+          source: '@ember/debug',
+          assertPredicateIndex: 1,
+        },
+      },
+      '@embroider/core:@ember/debug stripping',
+    ],
```

Comment:
Now that i re-enabled this block, I remember why I disabled it.

This block makes tests pass, but trips up against the globals warnings. Removing it makes some tests fail, especially when add-ons reference things like `@glimmer/tracking`.  There are several approaches we can take to address this, I hope to sync up with @ef4 tomorrow to discuss options.